File: packages/ts-client-sdk/src/sdk/models/operations/queryproductsbyid.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";


export class QueryProductsByIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=productId" })
  productId: string;
}

export class QueryProductsByIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: QueryProductsByIdPathParams;
}

export class QueryProductsByIdDefaultApplicationJSONIssues extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=message" })
  message: string;
}

export class QueryProductsByIdDefaultApplicationJSON extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=code" })
  code: string;

  @SpeakeasyMetadata({ data: "json, name=issues", elemType: QueryProductsByIdDefaultApplicationJSONIssues })
  issues?: QueryProductsByIdDefaultApplicationJSONIssues[];

  @SpeakeasyMetadata({ data: "json, name=message" })
  message: string;
}
export enum QueryProductsById200ApplicationJSONSeatingConfigSeatingStrategyNameEnum {
    MonthlyActiveUser = "monthly_active_user",
    FirstComeFirstServed = "first_come_first_served"
}

export class QueryProductsById200ApplicationJSONSeatingConfig extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=default_low_seat_warning_level_percent" })
  defaultLowSeatWarningLevelPercent: number;

  @SpeakeasyMetadata({ data: "json, name=default_seat_expiry_in_days" })
  defaultSeatExpiryInDays?: number;

  @SpeakeasyMetadata({ data: "json, name=limited_overflow_seating_enabled" })
  limitedOverflowSeatingEnabled: boolean;

  @SpeakeasyMetadata({ data: "json, name=low_seat_warning_level_pct" })
  lowSeatWarningLevelPct: number;

  @SpeakeasyMetadata({ data: "json, name=owner_id" })
  ownerId: string;

  @SpeakeasyMetadata({ data: "json, name=seat_reservation_expiry_in_days" })
  seatReservationExpiryInDays?: number;

  @SpeakeasyMetadata({ data: "json, name=seating_strategy_name" })
  seatingStrategyName: QueryProductsById200ApplicationJSONSeatingConfigSeatingStrategyNameEnum;
}

export class QueryProductsById200ApplicationJSON extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=contact_page_url" })
  contactPageUrl: string;

  @SpeakeasyMetadata({ data: "json, name=contact_sales_email" })
  contactSalesEmail: string;

  @SpeakeasyMetadata({ data: "json, name=contact_sales_url" })
  contactSalesUrl: string;

  @SpeakeasyMetadata({ data: "json, name=contact_support_email" })
  contactSupportEmail: string;

  @SpeakeasyMetadata({ data: "json, name=contact_support_url" })
  contactSupportUrl: string;

  @SpeakeasyMetadata({ data: "json, name=home_page_url" })
  homePageUrl: string;

  @SpeakeasyMetadata({ data: "json, name=id" })
  id: string;

  @SpeakeasyMetadata({ data: "json, name=is_setup_complete" })
  isSetupComplete: boolean;

  @SpeakeasyMetadata({ data: "json, name=on_access_denied_url" })
  onAccessDeniedUrl: string;

  @SpeakeasyMetadata({ data: "json, name=on_access_granted_url" })
  onAccessGrantedUrl: string;

  @SpeakeasyMetadata({ data: "json, name=on_no_seat_available_url" })
  onNoSeatAvailableUrl: string;

  @SpeakeasyMetadata({ data: "json, name=on_no_subscriptions_found_url" })
  onNoSubscriptionsFoundUrl: string;

  @SpeakeasyMetadata({ data: "json, name=on_subscription_canceled_url" })
  onSubscriptionCanceledUrl: string;

  @SpeakeasyMetadata({ data: "json, name=on_subscription_not_found_url" })
  onSubscriptionNotFoundUrl: string;

  @SpeakeasyMetadata({ data: "json, name=on_subscription_not_ready_url" })
  onSubscriptionNotReadyUrl: string;

  @SpeakeasyMetadata({ data: "json, name=on_subscription_suspended_url" })
  onSubscriptionSuspendedUrl: string;

  @SpeakeasyMetadata({ data: "json, name=owner_id" })
  ownerId: string;

  @SpeakeasyMetadata({ data: "json, name=privacy_notice_page_url" })
  privacyNoticePageUrl: string;

  @SpeakeasyMetadata({ data: "json, name=product_name" })
  productName: string;

  @SpeakeasyMetadata({ data: "json, name=publisher_name" })
  publisherName: string;

  @SpeakeasyMetadata({ data: "json, name=seatingConfig" })
  seatingConfig: QueryProductsById200ApplicationJSONSeatingConfig;
}

export class QueryProductsByIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  queryProductsById200ApplicationJSONObject?: QueryProductsById200ApplicationJSON;

  @SpeakeasyMetadata()
  queryProductsByIdDefaultApplicationJSONObject?: QueryProductsByIdDefaultApplicationJSON;
}

File: packages/go-client-sdk/pkg/models/operations/queryproductsbyid.go
package operations

type QueryProductsByIDPathParams struct {
	ProductID string `pathParam:"style=simple,explode=false,name=productId"`
}

type QueryProductsByIDRequest struct {
	PathParams QueryProductsByIDPathParams
}

type QueryProductsByIDDefaultApplicationJSONIssues struct {
	Message string `json:"message"`
}

type QueryProductsByIDDefaultApplicationJSON struct {
	Code    string                                          `json:"code"`
	Issues  []QueryProductsByIDDefaultApplicationJSONIssues `json:"issues,omitempty"`
	Message string                                          `json:"message"`
}

type QueryProductsByID200ApplicationJSONSeatingConfigSeatingStrategyNameEnum string

const (
	QueryProductsByID200ApplicationJSONSeatingConfigSeatingStrategyNameEnumMonthlyActiveUser    QueryProductsByID200ApplicationJSONSeatingConfigSeatingStrategyNameEnum = "monthly_active_user"
	QueryProductsByID200ApplicationJSONSeatingConfigSeatingStrategyNameEnumFirstComeFirstServed QueryProductsByID200ApplicationJSONSeatingConfigSeatingStrategyNameEnum = "first_come_first_served"
)

type QueryProductsByID200ApplicationJSONSeatingConfig struct {
	DefaultLowSeatWarningLevelPercent float64                                                                 `json:"default_low_seat_warning_level_percent"`
	DefaultSeatExpiryInDays           *float64                                                                `json:"default_seat_expiry_in_days,omitempty"`
	LimitedOverflowSeatingEnabled     bool                                                                    `json:"limited_overflow_seating_enabled"`
	LowSeatWarningLevelPct            float64                                                                 `json:"low_seat_warning_level_pct"`
	OwnerID                           string                                                                  `json:"owner_id"`
	SeatReservationExpiryInDays       *float64                                                                `json:"seat_reservation_expiry_in_days,omitempty"`
	SeatingStrategyName               QueryProductsByID200ApplicationJSONSeatingConfigSeatingStrategyNameEnum `json:"seating_strategy_name"`
}

type QueryProductsByID200ApplicationJSON struct {
	ContactPageURL             string                                           `json:"contact_page_url"`
	ContactSalesEmail          string                                           `json:"contact_sales_email"`
	ContactSalesURL            string                                           `json:"contact_sales_url"`
	ContactSupportEmail        string                                           `json:"contact_support_email"`
	ContactSupportURL          string                                           `json:"contact_support_url"`
	HomePageURL                string                                           `json:"home_page_url"`
	ID                         string                                           `json:"id"`
	IsSetupComplete            bool                                             `json:"is_setup_complete"`
	OnAccessDeniedURL          string                                           `json:"on_access_denied_url"`
	OnAccessGrantedURL         string                                           `json:"on_access_granted_url"`
	OnNoSeatAvailableURL       string                                           `json:"on_no_seat_available_url"`
	OnNoSubscriptionsFoundURL  string                                           `json:"on_no_subscriptions_found_url"`
	OnSubscriptionCanceledURL  string                                           `json:"on_subscription_canceled_url"`
	OnSubscriptionNotFoundURL  string                                           `json:"on_subscription_not_found_url"`
	OnSubscriptionNotReadyURL  string                                           `json:"on_subscription_not_ready_url"`
	OnSubscriptionSuspendedURL string                                           `json:"on_subscription_suspended_url"`
	OwnerID                    string                                           `json:"owner_id"`
	PrivacyNoticePageURL       string                                           `json:"privacy_notice_page_url"`
	ProductName                string                                           `json:"product_name"`
	PublisherName              string                                           `json:"publisher_name"`
	SeatingConfig              QueryProductsByID200ApplicationJSONSeatingConfig `json:"seatingConfig"`
}

type QueryProductsByIDResponse struct {
	ContentType                                   string
	StatusCode                                    int64
	QueryProductsByID200ApplicationJSONObject     *QueryProductsByID200ApplicationJSON
	QueryProductsByIDDefaultApplicationJSONObject *QueryProductsByIDDefaultApplicationJSON
}
